plugins/lang/python/python_loader.cc

```diff
@@ -634,6 +634,47 @@ class mldb_wrapper(object):
             if res.wasSuccessful():
                 self.script.set_return("success")
 
+class MldbUnitTest(unittest.TestCase):
+    def _assert_flat_result(self, res, expected):
+        if res[0][0] != '_rowName':
+            raise Exception("Cannot order rows if header is missing from "
+                            "response")
+        if expected[0][0] != '_rowName':
+            raise Exception("The first row of expected needs to be the header")
+        res_keys = sorted(res[0])
+        expected_keys = sorted(expected[0])
+        self.assertEqual(res_keys, expected_keys)
+
+        expected_order = {
+            key: index for index, key in enumerate(expected[0])}
+
+        reorder_directives = [expected_order[key] for key in res[0]]
+        ordered_res = []
+        for v in res[1:]:
+            ordered_res.append([v[pos] for pos in reorder_directives])
+        ordered_res = sorted(ordered_res)
+        expected = sorted(expected[1:])
+        for res_row, expected_row in zip(ordered_res, expected):
+            self.assertEqual(res_row, expected_row)
+
+    def _assert_json_result(self, res, expected):
+        res = sorted(res, key=lambda(v): v['rowName'])
+        expected = sorted(expected, key=lambda(v): v['rowName'])
+        for res_row, expected_row in zip(res, expected):
+            self.assertEqual(res_row["rowName"], expected_row["rowName"])
+            res_columns = sorted(res_row["columns"])
+            expected_columns = sorted(expected_row["columns"])
+            self.assertEqual(res_columns, expected_columns)
+
+
+    def assertQueryResult(self, res, expected):
```

Comment:
This is the new function. Works both with `mldb.query` and `mldb.get("/v1/query", ...).json()`